### PR TITLE
Adding null IdentityLogger to prevent null reference when using cache logger

### PR DESCRIPTION
**Pull request by trwalke**
Fixes #
Prevents null reference when no logger is set on the client app and logging is enabled on the token cache notification args

**Changes proposed in this request**
Adding null IdentityLogger to null logger

**Testing**
Unit test

**Performance impact**
N/A

**Documentation**
- [x] All relevant documentation is updated.

### CODE REVIEW

**Review by neha-bhargava — APPROVED** [review]
Looks good

File: src/client/Microsoft.Identity.Client/Internal/Logger/NullIdentityLogger.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Principal;
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/Internal/Logger/NullIdentityLogger.cs:7]: you can remove unused

**Review by gladjohn — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/Internal/Logger/NullLogger.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ internal class NullLogger : ILoggerAdapter
 
         public bool IsDefaultPlatformLoggingEnabled { get; } = false;
 
-        public IIdentityLogger IdentityLogger { get; } = null;
+        public IIdentityLogger IdentityLogger { get; } = new NullIdentityLogger();
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Internal/Logger/NullLogger.cs:24]: nit: Can a NullIdentityLogger from the IdentityModel abstractions package be used here?

**Review by pmaytak — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/Internal/Logger/NullIdentityLogger.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.IdentityModel.Abstractions;
+
+namespace Microsoft.Identity.Client.Internal.Logger
+{
+    internal class NullIdentityLogger : IIdentityLogger
```
Comment (jmprieur) [inline src/client/Microsoft.Identity.Client/Internal/Logger/NullIdentityLogger.cs:8]: Shouldn't it be provided in Microsoft.IdentityModel.Logging.Abstractions?

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @trwalke

File: src/client/Microsoft.Identity.Client/Internal/Logger/NullLogger.cs (hunk 2 above)
Comment (trwalke) [reply src/client/Microsoft.Identity.Client/Internal/Logger/NullLogger.cs:24]: Yes, didnt realize it existed.

File: src/client/Microsoft.Identity.Client/Internal/Logger/NullIdentityLogger.cs (hunk 3 above)
Comment (trwalke) [reply src/client/Microsoft.Identity.Client/Internal/Logger/NullIdentityLogger.cs:8]: Yes, didnt realize it existed.

**Pull request merged**